### PR TITLE
chore: add telemetry for examples

### PR DESCRIPTION
**Pull request by cdrage**
chore: add telemetry for examples

### What does this PR do?

Adds telemetry for building as well as pulling examples

### Screenshot / video of UI

N/A

### What issues does this PR fix or reference?

Closes https://github.com/podman-desktop/extension-bootc/issues/1047

### How to test this PR?

Functions on example should work as normal.

Signed-off-by: Charlie Drage <charlie@charliedrage.com>

### CODE REVIEW

**Comment by cdrage** [comment]
@deboer-tim @slemeur 

If you can double-check that I did this right, please let me know! First time adding telemetry.

File: packages/frontend/src/lib/ExampleCard.spec.ts [hunk 1]
```diff
@@ -90,10 +91,16 @@ test('pullImage function is called when Pull image button is clicked', async ()
 
   // Find and click the "Pull image" button
   const pullButton = screen.getByTitle('Pull image');
+
+  // spy on telemetryLogUsage function
+  const spyOnLogUsage = vi.spyOn(bootcClient, 'telemetryLogUsage');
```
Comment (axel7083) [inline packages/frontend/src/lib/ExampleCard.spec.ts:96]: You can probably do `vi.mocked(bootcClient.telemetryLogUsage)` and `expect(bootcClient.telemetryLogUsage).toHaveBeenCalled()`

File: packages/frontend/src/lib/ExampleCard.spec.ts [hunk 2]
```diff
@@ -90,10 +91,16 @@ test('pullImage function is called when Pull image button is clicked', async ()
 
   // Find and click the "Pull image" button
   const pullButton = screen.getByTitle('Pull image');
+
+  // spy on telemetryLogUsage function
+  vi.mocked(bootcClient.telemetryLogUsage);
```
Comment (jeffmaury) [inline packages/frontend/src/lib/ExampleCard.spec.ts:96]: I don't get why this is required ?

**Comment by cdrage** [comment]
Merging this in as we have 1 approval and code reviews too.

**Pull request merged**